cupy/cusparse.py

```diff
@@ -45,6 +52,158 @@ def _call_cusparse(name, dtype, *args):
     return f(*args)
 
 
+def csrmv(a, x, y=None, alpha=1, beta=0, transa=False):
+    """Matrix-vector product for a CSR-matrix and a dense vector.
+
+    .. math::
+
+       y = \\alpha * o_a(A) x + \\beta y,
+
+    where :math:`o_a` is a transpose function when ``transa`` is ``True`` and
+    is an identity function otherwise.
+
+    Args:
+        a (cupy.cusparse.csr_matrix): Matrix A.
+        x (cupy.ndarray): Vector x.
+        y (cupy.ndarray or None): Vector y.
+        alpha (float): Coefficient for x.
+        beta (float): Coefficient for y.
+        transa (bool): If ``True``, transpose of ``A`` is used.
+
+    Returns:
+        cupy.ndarray: Calculated ``y``.
+
+    """
+    if a.shape[1] != len(x):
+        raise ValueError('dimension mismatch')
+    a_shape = a.shape if not transa else a.shape[::-1]
+
+    handle = device.get_cusparse_handle()
+    m, n = a_shape
+    a, x, y = _cast_common_type(a, x, y)
+    dtype = a.dtype
+    if y is None:
+        y = cupy.zeros(m, dtype)
+    else:
+        y = cupy.asfortranarray(y)
+    alpha = numpy.array(alpha, dtype).ctypes
+    beta = numpy.array(beta, dtype).ctypes
+    _call_cusparse(
+        'csrmv', dtype,
+        handle, _transpose_flag(transa),
+        m, n, a.nnz, alpha.data, a._descr.descriptor,
+        a.data.data.ptr, a.indptr.data.ptr, a.indices.data.ptr,
+        x.data.ptr, beta.data, y.data.ptr)
+
+    return y
+
+
+def csrmm(a, b, c=None, alpha=1, beta=0, transa=False):
+    """Matrix-matrix product for a CSR-matrix and a dense matrix.
+
+    .. math::
+
+       C = \\alpha o_a(A) B + \\beta C,
+
+    where :math:`o_a` is a transpose function when ``transa`` is ``True`` and
+    is an identity function otherwise.
+
+    Args:
+        a (cupy.sparse.csr): Sparse matrix A.
+        b (cupy.ndarray): Dense matrix B. It must be F-contiguous.
```

Comment:
Requiring F-contiguousness as the function's spec? `cupy.asfortranarray()` is called inside.